src/cgroups.c

```diff
@@ -219,38 +292,34 @@ static int cgroups_read (void)
 
 	for (mnt_ptr = mnt_list; mnt_ptr != NULL; mnt_ptr = mnt_ptr->next)
 	{
-		/* Find the cgroup mountpoint which contains the cpuacct
-		 * controller. */
-		if ((strcmp(mnt_ptr->type, "cgroup") != 0)
-				|| !cu_mount_checkoption(mnt_ptr->options,
-					"cpuacct", /* full = */ 1))
+		if (strcmp(mnt_ptr->type, "cgroup") != 0)
 			continue;
 
-		walk_directory (mnt_ptr->dir, read_cpuacct_root,
-				/* user_data = */ NULL,
-				/* include_hidden = */ 0);
-		cgroup_found = 1;
-		/* It doesn't make sense to check other cpuacct mount-points
-		 * (if any), they contain the same data. */
-		break;
+		found_all = 1;
+		for (i = 0; i < STATIC_ARRAY_SIZE(settings); i++) {
+			cgroups_walk_mountpoint(mnt_ptr, &settings[i]);
+			found_all &= settings[i].found;
+		}
+
+		if (found_all)
+			break;
 	}
 
 	cu_mount_freelist (mnt_list);
 
-	if (!cgroup_found)
-	{
-		WARNING ("cgroups plugin: Unable to find cgroup "
-				"mount-point with the \"cpuacct\" option.");
-		return (-1);
-	}
+	for (i = 0; i < STATIC_ARRAY_SIZE(settings); i++)
+		if (0 == settings[i].found)
+			WARNING ("cgroups plugin: Unable to find cgroup "
+				 "mount-point with the \"%s\" option.", settings[i].controller);
 
-	return (0);
+	return found_all ? 0 : -1;
```

Comment:
If the memory controller is not available (like on default Debian 8 kernels without cgroup_enable=memory) the read function now fails and gets suspended. Before this change, you would get the cpuacct metrics.

I wonder if it would be better to log a warning but return success here if the memory controller is not available.